### PR TITLE
ethercat-sys: generate ioctl numbers from master header

### PR DESCRIPTION
**Pull request by birkenfeld**

### CODE REVIEW

**Comment by flosse** [comment]
Awesome!
I'll test it soon and merge it afterwards :)

**Comment by birkenfeld** [comment]
Unfortunately the `none` vs `arg` distinction that is only in the Rust version needs manual maintenance of the APIs that need `arg` for `EC_IO()`.  Would it be better to have to explicitly list both and error out if a hitherto unknown ioctl is found?

**Review by flosse — APPROVED** [review]

File: ethercat-sys/build.rs [hunk 1]
```diff
@@ -33,40 +30,59 @@ fn main() {
...
+            // So as a first dirty workaround we just ignore them
+            // to be able to compile.
+            if name.starts_with("64_REF_CLK") {
+                continue;
+            }
+
             let mut numparts = parts[2].split("(");
             let access = match numparts.next().unwrap() {
                 "EC_IO" => match name {
                     "SEND" => "arg",
```
Comment (birkenfeld) [inline ethercat-sys/build.rs]: I noticed some more `arg` in your port for sncn-11, do you want to add some?

File: ethercat-sys/build.rs [hunk 2]
```diff
@@ -33,40 +30,59 @@ fn main() {
...
+
+            // FIXME:
+            // There are constant names that start with a number
+            // so they would be invalid rust code.
+            // The problem exists e.g. if you try to
+            // compile the v1.5.2-sncn-11 branch of the synapticon
+            // fork: https://github.com/synapticon/Etherlab_EtherCAT_Master
+            // So as a first dirty workaround we just ignore them
+            // to be able to compile.
+            if name.starts_with("64_REF_CLK") {
```
Comment (birkenfeld) [inline ethercat-sys/build.rs:53]: just add "_" in front?

File: ethercat-sys/build.rs (hunk 1 above)
Comment (flosse) [reply ethercat-sys/build.rs]: Good point! I'll hopefully have a look at it tomorrow :)

**Pull request merged**